server/kitsu/addon_helpers.py

```diff
@@ -0,0 +1,95 @@
+import unicodedata
+import re
+from nxtools import slugify
+
+"""
+A collection of helper functions for Ayon Addons
+minimal dependencies, pytest unit tests
+"""
+
+
+## ========== KITSU -> AYON NAME CONVERSIONS =====================
+
+
+def create_short_name(name: str) -> str:
+    """create a shortname from the full name when a shortname is not present"""
+    code = name.lower()
+
+    if "_" in code:
+        subwords = code.split("_")
+        code = "".join([subword[0] for subword in subwords])[:4]
+    elif len(name) > 4:
+        vowels = ["a", "e", "i", "o", "u"]
+        filtered_word = "".join([char for char in code if char not in vowels])
+        code = filtered_word[:4]
+
+    # if there is a number at the end of the code, add it to the code
+    last_char = code[-1]
+    if last_char.isdigit():
+        code += last_char
+
+    return code
+
+
+def to_username(first_name: str, last_name: str | None = None) -> str:
+    """converts usernames from kitsu - converts accents"""
+
+    name = (
+        f"{first_name.strip()}.{last_name.strip()}" if last_name else first_name.strip()
+    )
+
+    name = name.lower()
+    name = remove_accents(name)
+    return to_entity_name(name)
+
+
+def remove_accents(input_str: str) -> str:
+    """swap accented characters for a-z equivilants ž => z"""
+
+    nfkd_form = unicodedata.normalize("NFKD", input_str)
+    result = "".join([c for c in nfkd_form if not unicodedata.combining(c)])
+
+    # manually replace exceptions
+    # @see https://stackoverflow.com/questions/3194516/replace-special-characters-with-ascii-equivalent
+    replacement_map = {
+        "Æ": "AE",
+        "Ð": "D",
+        "Ø": "O",
+        "Þ": "TH",
+        "ß": "ss",
+        "æ": "ae",
+        "ð": "d",
+        "ø": "o",
+        "þ": "th",
+        "Œ": "OE",
+        "œ": "oe",
+        "ƒ": "f",
+    }
+    for k, v in replacement_map.items():
+        if k in result:
+            result = result.replace(k, v)
+
+    # remove any unsupported characters
+    result = re.sub(r"[^a-zA-Z0-9_\.\-]", "", result)
+
+    return result
+
+
+def to_entity_name(name) -> str:
+    """convert names so they will pass Ayon Entity name validation
+    @see ayon_server.types.NAME_REGEX = r"^[a-zA-Z0-9_]([a-zA-Z0-9_\.\-]*[a-zA-Z0-9_])?$"
+    """
+
+    assert name, "Entity name cannot be empty"
```

Comment:
How this can happen?
---
AssertionError is for testing modules, other exception should be raise.
```suggestion
    if not name:
        raise ValueError("Entity name cannot be empty")
```

Reply:
There are asserts in the rest of the addon code so I thought it was OK, but ValueError is much better thanks!